src/EventHandling/DelegateEventHandlingToSpecificMethodTrait.php

```diff
@@ -37,7 +37,7 @@ private function getHandleMethodName($event): ?string
 
         $expectedClass = $parameter->getClass();
 
-        if ($expectedClass->getName() !== get_class($event)) {
+        if ($expectedClass !== null && $expectedClass->getName() !== get_class($event)) {
```

Comment:
If `$expectedClass` is `null` should this method not return `null`?

Reply:
Changed it into an or